### DISCUSSION TITLE
Cronjobs, or server functions...

### DISCUSSION THREAD

**Discussion by Musiphilos** [coollabsio/coolify#3212] [Q&A]
Hi alll,

I am looking to set up a process that runs code at regular, time-based intervals, specifically to send API requests to external services on an automatic and recurring basis.

Is there built-in support in Coolify for cron jobs or similar server functions that allow for this kind of scheduling? If so, could you guide me on how to configure it properly? If not, is there a recommended approach for integrating external cron-like services to achieve this?

Thank you for your help!

**Comment by Vahor** [comment]
You can use `Scheduled Tasks`
![image](https://github.com/user-attachments/assets/6b1905c3-25f5-404c-962f-ca9cadc7edaf)

![image](https://github.com/user-attachments/assets/708cbd71-d491-4171-a6a8-b34bd8a67276)

And run your command there. like curl for external services, and you also have access to env vars

**Comment by khashashin** [comment]
I recently searched for a scheduler on the server and found one at the project level. The problem is that this scheduler runs scripts within a container, so you have to select the name of the container. I would like to have a scheduler at the server level, i.e., one that runs scripts within the server instance itself and not within the container.

**Reply by bodoczky** [reply]
> I recently searched for a scheduler on the server and found one at the project level.

what exactly have you found? You found a resource that we can add or are you talking about some custom thing that you have on the server in question?

**Reply by khashashin** [reply]
Sorry, maybe my message wasn't clear. I meant the existing scheduler in Coolify.

<img width="1915" height="906" alt="image" src="https://github.com/user-attachments/assets/96e7ad3a-cddb-4384-a38b-e753f5d4d5d5" />

**Reply by dylanmorroll** [reply]
I think I am also interested in this if I understand correctly. So I have project that includes a database. I want to run some cronjobs to ingest data into that db. However, the 'Scheduled Tasks' option is only available once you have created a New Resource/Application.

I presume the implication of this is you have to create an Application on-top of your Git repo, creating a docker container in your server, that runs cronjobs. To me it would make more sense to run them at the server level, rather than within a docker container?

**Reply by av1m** [reply]
You are both completely right. Coolify’s built-in **Scheduled Tasks** are strictly container-scoped right now—under the hood, Coolify just runs a `docker exec` inside that specific application's container.

Since a native server-level scheduler isn't available in the UI yet, here are the two best ways to make this work:

### Option 1: Use the Host Server's Native Cron (easiest for server-level)

Since Coolify runs on a standard Linux VPS, you can bypass the UI entirely and use the server's own crontab via SSH:

1. SSH into your server and open the cron editor: `crontab -e`
2. Add your cron job. For example, to run a host script that interacts with a Coolify database container:
```bash
0 * * * * docker exec -i <db_container_name> mysql -u user -p db_name < /path/to/host/script.sql

```

### Option 2: The UI Workaround (Docker Socket Mount)

If you want to keep everything manageable inside the Coolify UI, you can deploy a lightweight "worker" app (like a basic Python/Alpine container) to handle your scripts.

To give this container server-level powers, go to its **Storage / Volumes** settings in Coolify and mount the host's Docker socket:

* **Source:** `/var/run/docker.sock`
* **Target:** `/var/run/docker.sock`

Now you can use Coolify's built-in Scheduled Tasks on this worker container, and your scripts will be able to talk to the host and interact with any other container on the server.

**Comment by DiogenesTSO** [comment]
Bom dia, aproveitando a discussão, como que voces rodam o queue:work? Tenho um Scheduled task no coolify onde executa um command no laravel e dispara jobs, quero deixar o worker sempre em standby pra quando executar a tarefas os jobs serem processados simultaneos. Agradeço

**Reply by clatorre-csti** [reply]
I think you can add a Redis container to your app as a Laravel queue drive
